<a href="https://colab.research.google.com/github/rizzken/PreAppointmentPatientSummary-Eval/blob/main/02_summary_evaluation.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
#Choose sample data for the further ML testing. Execute only once to create a file with the samples
import pandas as pd
from google.colab import drive
drive.mount('/content/drive')

df = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/patients_with_age.csv')

samples = df.sample(10)
samples.to_csv('patients_samples.csv', index=False)
print("File saved: patients_samples.csv")

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).
File savec: patients_samples1.csv


In [ ]:
#Read from new samples file. Index was added manually. Woopsie :)
import pandas as pd
from google.colab import drive

df = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/patients_samples1.csv')
print(df)

   Unnamed: 0                                    id   birthdate   deathdate  \
0           1  02bdcf90-32f9-f93f-79d8-5a1933f2af06  1992-09-13  1958-05-21   
1           2  f70a6900-4871-11c4-cbb2-4ee86975558c  1960-01-06         NaN   
2           3  17d6e339-f882-e88c-320e-c3f71c62b945  1993-06-27         NaN   
3           4  43040569-f355-dcc7-1fa0-c2d386a57cc7  2007-02-18         NaN   
4           5  d74d24e5-192e-03f7-5bda-6cd8195bbefe  2011-09-16         NaN   
5           6  fb1ff537-0e7d-77ed-b2be-3cbd195cc64b  2004-03-24         NaN   
6           7  d70a6bbf-8ebd-77e0-2f04-d3369c9f0fcb  1963-04-21         NaN   
7           8  db4ce617-081b-c16a-270f-8f8391bee5d3  2006-09-25         NaN   
8           9  b2ec1b43-0b3b-eec8-e2ec-21d19a1033d6  2002-01-06         NaN   
9          10  c5a11ac2-b86c-7f70-de60-5daed3edc3d9  1953-03-05         NaN   

           ssn     drivers    passport prefix        first           last  \
0  999-29-6086  99938340.0  X84875389X    Mr.    Rich

###1. Patient ID: 02bdcf90-32f9-f93f-79d8-5a1933f2af06
<details>
<summary>Original Record (click to expand)</summary>

**Full record:**  
- Name: Mr. Richie600 McCullough561
- Marital status: Married
- Race: white,nonhispanic
- Gender: male  
- Birthdate: 1992-09-13  
- Deathdate: 1958-05-21 (should be replaced with NaN)  
- Age: (not calculated due to invalid deathdate)  
- SSN: 999-29-6086  
- Passport: X84875389X  
- Drivers: 99938340  
- Birthplace Daphne, Alabama, US  
- Address: 912 Goyette Ranch  
- City: Center Point  
- State: Alabama  
- County: Jefferson county  
- Fips: 1073  
- Zip: 35215  
- Helthcare expenses: Unknown (missing)
- Healthcare coverage: 14104.06   
- Income: 30043.0  
- Location: POINT(-86.60579960825079 33.668945899945584)

</details>

**Manual Summary (prepared by QA)**

**Patient**: Mr. Richie600 McCullough561, male  
**Age**: Not calculated (invalid deathdate)  
**Marital status**: Married  
**Location**: Center Point, Alabama, US
**Income**: 30,043 per year  
**Healthcare coverage**: 14,104.06  
**Healthcare expenses**: Unknown (missing)  
**Key notes**:  
- Invalid deathdate (earlier than birthdate) — data error  
- No medical history, allergies or procedures in demographics  
- SSN, Passport and Drivers present — full identification available

**Quick Evaluation**  
- Coverage: partial  
- Hallucinations: none  
- Factual errors: yes  
- Score: 3/5

###2. Patient ID: f70a6900-4871-11c4-cbb2-4ee86975558c
<details>
<summary>Original Record (click to expand)</summary>

**Full record:**  
- Name: Mrs. Scarlett814 Conroy74
- Marital status: Married
- Race: black,nonhispanic
- Gender: female  
- Birthdate: 1960-01-06  
- Deathdate:  Unknown (missing)
- Age: 66  
- SSN: 999-16-2177  
- Passport: X77565933X  
- Drivers: 99966246  
- Birthplace: Cullman  Alabama  US  
- Address: 130 Goyette Gate Suite 92
- City: Atmore  
- State: Alabama  
- County: Escambia County
- Fips: 1053  
- Zip: 36502  
- Healthcare expenses: 1767497.44
- Healthcare coverage: 25482.44  
- Income: Unknown (missing)  
- Location: POINT(-87.47009719785575 31.11588851471998)

</details>

**Manual Summary (prepared by QA)**

**Patient**: Mrs. Scarlett814 Conroy74, female  
**Age**: 66  
**Marital status**: Married  
**Location**: Atmore, Alabama, US
**Income**: Unknown (missing)
**Healthcare coverage**: 25482.44   
**Healthcare expenses**: 1767497.44  
**Key notes**:   
- No medical history, allergies or procedures in demographics  
- SSN, Passport and Drivers present — full identification available

**Quick Evaluation**  
- Coverage: partial  (might be unemployed)
- Hallucinations: none  
- Factual errors: yes  
- Score: 4/5

###3. Patient ID: 17d6e339-f882-e88c-320e-c3f71c62b945
<details>
<summary>Original Record (click to expand)</summary>

**Full record:**  
- Name: Mrs. Blanca837 Ontiveros947
- Marital status: Unknown (missing)
- Race: black,nonhispanic
- Gender: female  
- Birthdate: 1993-06-27  
- Deathdate:  Unknown (missing)
- Age: 32  
- SSN: 999-55-9208  
- Passport: X80214132X  
- Drivers: 99952359  
- Birthplace: Wesley  Saint Andrew Parish  DM  
- Address: 303 Klocko Underpass Apt 78
- City: Harvest  
- State: Alabama  
- County: Madison County  
- Fips: 1089  
- Zip: 35749  
- Healthcare expenses: 66794.92
- Healthcare coverage: 386110.45  
- Income: 23803.0  
- Location: POINT(-86.7686032209981 34.80024351735981)

</details>

**Manual Summary (prepared by QA)**

**Patient**: Mrs. Blanca837 Ontiveros947, female  
**Age**: 32  
**Marital status**: Unknown (missing)  
**Location**: Harvest, Alabama, US
**Income**: 23803.0
**Healthcare coverage**: 386110.45   
**Healthcare expenses**: 66794.92  
**Key notes**:   
- No medical history, allergies or procedures in demographics  
- SSN, Passport and Drivers present — full identification available

**Quick Evaluation**  
- Coverage: partial  (marital status is missing)
- Hallucinations: none  
- Factual errors: none  
- Score: 4/5

###4. Patient ID: 43040569-f355-dcc7-1fa0-c2d386a57cc7
<details>
<summary>Original Record (click to expand)</summary>

**Full record:**  
- Name: Lila466 Purdy2
- Marital status: Unknown (missing)
- Race: white,nonhispanic
- Gender: female  
- Birthdate: 2007-02-18  
- Deathdate:  Unknown (missing)
- Age: 19  
- SSN: 999-82-9397  
- Passport: Unknown (missing)  
- Drivers: Unknown (missing)  
- Birthplace: Homewood  Alabama  US  
- Address: 906 Grimes Corner
- City: Alexander city  
- State: Alabama  
- County: Tallapoosa County  
- Fips: 1037  
- Zip: 35089  
- Healthcare expenses: 19500.93
- Healthcare coverage: 7961.73  
- Income: 38626.0  
- Location: POINT(-85.92242795595156 32.8836308269105)

</details>

**Manual Summary (prepared by QA)**

**Patient**: Lila466 Purdy2, female  
**Age**: 19  
**Marital status**: Unknown (missing)  
**Location**: Alexander city, Alabama, US
**Income**: 38626.0
**Healthcare coverage**: 7961.73   
**Healthcare expenses**: 19500.93  
**Key notes**:   
- No medical history, allergies or procedures in demographics  
- Passport and Drivers missing, but SSN presents — at least one ID document should be enough

**Quick Evaluation**  
- Coverage: partial  (prefix and marital status missing from source data)
- Hallucinations: none  
- Factual errors: none  
- Score: 4/5

###5. Patient ID: d74d24e5-192e-03f7-5bda-6cd8195bbefe
<details>
<summary>Original Record (click to expand)</summary>

**Full record:**  
- Name: Fidelia216 Flatley871
- Marital status: Unknown (missing)
- Race: white,nonhispanic
- Gender: female  
- Birthdate: 2011-09-16  
- Deathdate:  Unknown (missing)
- Age: 14  
- SSN: 999-76-3476  
- Passport: Unknown (missing)  
- Drivers: Unknown (missing)  
- Birthplace: Andalusia  Alabama  US  
- Address: 134 Kertzmann Overpass Unit 92
- City: Chelsea  
- State: Alabama  
- County: Shelby County  
- Fips: 1117  
- Zip: 35043  
- Healthcare expenses: 11038.10
- Healthcare coverage: Unknown (missing)  
- Income: Unknown (missing)  
- Location: POINT(-86.57248588048783 33.16112926687233)

</details>

**Manual Summary (prepared by QA)**

**Patient**: Lila466 Purdy2, female  
**Age**: 19  
**Marital status**: Unknown (missing)  
**Location**: Alexander city, Alabama, US
**Income**: 38626.0
**Healthcare coverage**: 7961.73   
**Healthcare expenses**: 19500.93  
**Key notes**:   
- No medical history, allergies or procedures in demographics  
- Passport and Drivers missing, but SSN present — at least one ID document should be enough

**Quick Evaluation**  
- Coverage: partial  (some data is missing from source data)
- Hallucinations: none  
- Factual errors: none  
- Score: 4/5

###6. Patient ID: fb1ff537-0e7d-77ed-b2be-3cbd195cc64b
<details>
<summary>Original Record (click to expand)</summary>

**Full record:**  
- Name: Mr. Kim439 Ferry570
- Marital status: Unknown (missing)
- Race: white,nonhispanic
- Gender: male  
- Birthdate: 2004-03-24  
- Deathdate:  Unknown (missing)
- Age: 21  
- SSN: 999-67-5361  
- Passport: Unknown (missing)  
- Drivers: 99921317  
- Birthplace: Smoke Rise  Alabama  US  
- Address: 296 Bruen Bridge
- City: Enterprise  
- State: Alabama  
- County: Coffee County  
- Fips: 1031  
- Zip: 36330    
- Healthcare expenses: Unknown (missing)
- Healthcare coverage: 0  
- Income: Unknown (missing)  
- Location: POINT(-85.89629086641092 31.389287147688766)

</details>

**Manual Summary (prepared by QA)**

**Patient**: Mr. Kim439 Ferry570, male  
**Age**: 21  
**Marital status**: Unknown (missing)  
**Location**: Enterprise, Alabama, US
**Income**: Unknown (missing)
**Healthcare coverage**: 0.0   
**Healthcare expenses**: Unknown (missing)  
**Key notes**:   
- No medical history, allergies or procedures in demographics  
- Passport is missing, but SSN and Drivers present — should be enough

**Quick Evaluation**  
- Coverage: partial  (some data is missing from source data)
- Hallucinations: none  
- Factual errors: none  
- Score: 4/5

###7. Patient ID: d70a6bbf-8ebd-77e0-2f04-d3369c9f0fcb
<details>
<summary>Original Record (click to expand)</summary>

**Full record:**  
- Name: Mr. Silas208 Schaefer657
- Marital status: married
- Race: black,nonhispanic
- Gender: male  
- Birthdate: 1963-04-21  
- Deathdate:  Unknown (missing)
- Age: 62  
- SSN: 999-61-6649  
- Passport: X49159141X  
- Drivers: 99957052  
- Birthplace: Hueytown  Alabama  US  
- Address: 408 Schultz Dale Suite 65
- City: Huntsville  
- State: Alabama  
- County: Madison County  
- Fips: 1089  
- Zip: 35758    
- Healthcare expenses: Unknown (missing)
- Healthcare coverage: 3397318.91  
- Income: 86178.0  
- Location: POINT(-86.4792298938542 34.66561730278715)

</details>

**Manual Summary (prepared by QA)**

**Patient**: Mr. Silas208 Schaefer657, male  
**Age**: 62  
**Marital status**: married  
**Location**: Huntsville, Alabama, US
**Income**: 86178.0
**Healthcare coverage**: 3397318.91   
**Healthcare expenses**: Unknown (missing)  
**Key notes**:   
- No medical history or allergies
- SSN, Passport and Drivers present — full identification available

**Quick Evaluation**  
- Coverage: full
- Hallucinations: none  
- Factual errors: none  
- Score: 5/5

###8. Patient ID: db4ce617-081b-c16a-270f-8f8391bee5d3
<details>
<summary>Original Record (click to expand)</summary>

**Full record:**  
- Name: Ms. Andrea7 Castañeda925
- Marital status: Unknown (missing)
- Race: white
- Gender: female  
- Birthdate: 2006-09-25  
- Deathdate:  Unknown (missing)
- Age: 19  
- SSN: 999-35-5482  
- Passport: Unknown (missing)  
- Drivers: 99965719  
- Birthplace: Panama City  Panama  PA  
- Address: 561 Crist Mill
- City: Decatur  
- State: Alabama  
- County: Morgan County  
- Fips: 1103  
- Zip: 35640    
- Healthcare expenses: 21092.48
- Healthcare coverage: 18608.37  
- Income: 197103.0  
- Location: POINT(-87.00281484510532 34.54188892087336)

</details>

**Manual Summary (prepared by QA)**

**Patient**: Ms. Andrea7 Castañeda925, female  
**Age**: 19  
**Marital status**: Unknown (missing)  
**Location**: Decatur, Alabama, US
**Income**: 197103.0
**Healthcare coverage**: 18608.37   
**Healthcare expenses**: 21092.48  
**Key notes**:   
- No medical history or allergies
- Passport is missing, but SSN and Drivers present — should be enough
- Suspicious income for a young age. Might be a mistake

**Quick Evaluation**  
- Coverage: partial  (some data is missing from source data)
- Hallucinations: none  
- Factual errors: special symbols in a full name  
- Score: 4/5

###9. Patient ID: b2ec1b43-0b3b-eec8-e2ec-21d19a1033d6
<details>
<summary>Original Record (click to expand)</summary>

**Full record:**  
- Name: Ms. Francene766 Bechtelar572
- Marital status: Unknown (missing)
- Race: white,nonhispanic
- Gender: female  
- Birthdate: 2002-01-06  
- Deathdate:  Unknown (missing)
- Age: 24  
- SSN: 999-19-9514  
- Passport: X9188549X  
- Drivers: 99936740  
- Birthplace: Montgomery  Alabama  US  
- Address: 403 Okuneva Path
- City: Montgomery  
- State: Alabama  
- County: Montgomery County  
- Fips: 1101  
- Zip: 36106    
- Healthcare expenses: 23217.41
- Healthcare coverage: 87559.79  
- Income: Unknown (missing)  
- Location: POINT(-86.27878257739646 32.43082472487542)

</details>

**Manual Summary (prepared by QA)**

**Patient**: Ms. Francene766 Bechtelar572, female  
**Age**: 24  
**Marital status**: Unknown (missing)  
**Location**: Montgomery, Alabama, US
**Income**: Unknown (missing)
**Healthcare coverage**: 87559.79   
**Healthcare expenses**: 23217.41  
**Key notes**:   
- No medical history or allergies
- SSN, Passport and Drivers present — full identification available
- Income unknown, might be unemployed

**Quick Evaluation**  
- Coverage: full
- Hallucinations: none  
- Factual errors: none  
- Score: 4/5

###10. Patient ID: c5a11ac2-b86c-7f70-de60-5daed3edc3d9
<details>
<summary>Original Record (click to expand)</summary>

**Full record:**  
- Name: Mrs. Ellena683 Pagac496
- Marital status: married
- Race: white,nonhispanic
- Gender: female  
- Birthdate: 1953-03-05  
- Deathdate:  Unknown (missing)
- Age: 73  
- SSN: 999-37-9914  
- Passport: X86621541X  
- Drivers: 99921198  
- Birthplace: Montgomery  Alabama  US  
- Address: 922 Hodkiewicz Branch
- City: Gadsden  
- State: Alabama  
- County: Etowah County  
- Fips: 1055  
- Zip: 35906    
- Healthcare expenses: 9999999.00
- Healthcare coverage: Unknown (missing)
- Income: 92831.0  
- Location: POINT(-85.99738594239358 34.065755516410874)

</details>

**Manual Summary (prepared by QA)**

**Patient**: Ms. Francene766 Bechtelar572, female  
**Age**: 73  
**Marital status**: married  
**Location**: Gadsden, Alabama, US
**Income**: 92831.0
**Healthcare coverage**: Unknown (missing)   
**Healthcare expenses**: 9999999.00  
**Key notes**:   
- No medical history or allergies
- SSN, Passport and Drivers present — full identification available
- Helthcare expenses look suspicious - might me an error

**Quick Evaluation**  
- Coverage: full (might have no insurance)
- Hallucinations: none  
- Factual errors: Helthcare expenses look suspicious  
- Score: 4/5

##General Evaluation Notes & Observations (after 10 samples)
Positive Findings
No hallucinations were detected in any of the 10 generated summaries. The model (QA :D) remained faithful to the provided source data.

###Key Issues
Incomplete coverage of critical clinical fields (medical history, allergies, diagnoses, and medications).
Invalid dates (one case with deathdate before birthdate).
Unrealistic financial values (e.g., expenses = 999999.00, zero income for working-age patients, and implausibly high income for 19-year-olds).
Some records require further clarification and data validation.

###Scoring Overview

**5/5**: 1 case  
**4/5**: 8 cases  
**3/5**: 1 case   

###Overall Assessment
The summaries are factually accurate and free of hallucinations, but their clinical usefulness is limited due to data quality issues and missing key medical information. Robust data validation and cleaning are recommended before any production use.